src/plugins/syscalls/syscalls.cpp

```diff
@@ -128,10 +130,103 @@ static event_response_t cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
     return 0;
 }
 
+static event_response_t win_cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+    int i,j,nargs;
+    unsigned char* buf; // pointer to buffer to hold argument values
+
+    syscall_wrapper_t *wrapper = (syscall_wrapper_t*)info->trap->data;
+    syscalls *s = wrapper->sc;
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    access_context_t ctx;
+    ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+    ctx.dtb = info->regs->cr3;
+
+    if(wrapper->syscall_index>-1) { // get arguments only if we know how many to get
+        nargs = win_syscall_struct[wrapper->syscall_index].num_args;
+
+        unsigned long size = s->reg_size * nargs;
+        buf = (unsigned char *)g_malloc(sizeof(char)*size);
+
+        if(s->reg_size==4){ // 32 bit os
+
+            ctx.addr = info->regs->rsp + s->reg_size;  // jump over base pointer
+
+            // multiply num args by 4 for 32 bit systems to get the number of bytes we need
+            // to read from the stack.  assumes standard calling convention (cdecl) for the
+            // visual studio compile.
+            if((size_t)size != vmi_read(vmi, &ctx, buf, size)){
+                goto exit;
+            }
+        }
+        else { // 64 bit os - ************** UNTESTED *******************!
+            buf[0] = info->regs->rcx;
```

Comment:
...but it might be better to simplify this whole part buy allocating `nargs*sizeof(addr_t)` to begin with and then we wouldn't have to do so much casting and weird loops and whatnot.